src/tests/timer-add-api-test/timer-add-api-test.c

```diff
@@ -0,0 +1,189 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "common_types.h"
+#include "osapi.h"
+#include "utassert.h"
+#include "uttest.h"
+#include "utbsp.h"
+
+#define NUMBER_OF_TIMERS 4
+#define TASK_1_ID         1
+#define TASK_1_STACK_SIZE 4096
+#define TASK_1_PRIORITY   101
+
+OS_time_t        StartTime;
+OS_time_t        EndTime;
+uint32           TimerStart[NUMBER_OF_TIMERS] = {1000, 2000000, 3000000, 4000000 };
+uint32           TimerInterval[NUMBER_OF_TIMERS] = {500000, 400000, 800000, 600000 };
+
+uint32 TimerTestTaskStack[TASK_1_STACK_SIZE];
+int32 timer_counter[NUMBER_OF_TIMERS];
+uint32 timer_idlookup[OS_MAX_TIMERS];
+
+/*
+** Test timer function.
+** Note: For some Host OSs, this is the equivalent of an ISR, so the calls available are limited.
+** For example, Linux and vxWorks can call functions like printf, but RTEMS cannot.
+*/
+void test_func(uint32 timer_id , void *arg)
+{
+   OS_ConvertToArrayIndex(timer_id, &timer_id);
+   timer_counter[timer_idlookup[timer_id]]++;
+}
```

Comment:
This should use the `arg` parameter as a pointer to the counter so it doesn't need to use the global/id workaround.  See other comment below.